scenes/globals/pause.gd

```diff
@@ -0,0 +1,91 @@
+extends Node
+
+enum System {
+	PLAYER_INPUT,
+	GAME
+	# For now is just these two but in the future we could have: Physics, Enemies, AI, etc
+}
+
+# Private vars
+var _pause_requests: Dictionary[System, Array] = {}
+
+# This should be an array of systems but it throws
+# a runtime error when using the filter function, so for now just "Array"
+var _paused_systems: Array = []
+
+
+# Public interface
+func pause_system(system: System, node: Node) -> void:
+	_add_request(system, node)
+
+
+func unpause_system(system: System, node: Node) -> void:
+	_clear_request(system, node)
+
+	_refresh_systems()
```

Comment:
This function is calling `_clear_request` which already calls `_refresh_systems()` at the end. So `_refresh_systems()` can be called twice. I would just move the content of `_clear_request` here and get rid of that function.